src/vmm/src/devices/virtio/block/device.rs

```diff
@@ -578,15 +578,20 @@ impl VirtioDevice for Block {
     }
 
     fn write_config(&mut self, offset: u64, data: &[u8]) {
-        let data_len = data.len() as u64;
-        let config_len = self.config_space.len() as u64;
-        if offset + data_len > config_len {
+        let Ok(start) = usize::try_from(offset) else {
```

Comment:
@00xc many thanks for your contribution!

Would you be able to adjust the implementation to avoid duplication of the error handling code, please:
```rust
    error!("Failed to write config space");
    METRICS.block.cfg_fails.inc();
```

Reply:
Done!